Description: use older major version of crate bytesize
Author: Jonas Smedegaard <dr@jones.dk>
Forwarded: not-needed
Last-Update: 2026-04-04
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -110,7 +110,7 @@
 anyhow = "1.0.58"
 arbitrary = "1.1.0"
 bitflags = "2.5.0"
-bytesize = "2.0.0"
+bytesize = "1.3.0"
 clap = { version = "4.0.0", features = ["derive"] }
 clap_complete = "4.4.7"
 comfy-table = { version = "7.1.3", default-features = false }
--- a/src/bin/wasm-tools/metadata.rs
+++ b/src/bin/wasm-tools/metadata.rs
@@ -165,8 +165,6 @@
         }
     };
     let size = ByteSize::b((range.end - range.start) as u64)
-        .display()
-        .si_short()
         .to_string();
 
     let usep = match ((range.end - range.start) as f64 / range_max as f64 * 100.0).round() as u8 {
--- a/tests/cli/metadata-add-component.wat.stdout
+++ b/tests/cli/metadata-add-component.wat.stdout
@@ -1,10 +1,10 @@
-╭───────────┬────────────┬──────┬───────┬───────────┬────────╮
-│ KIND      ┆ NAME       ┆ SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT │
-╞═══════════╪════════════╪══════╪═══════╪═══════════╪════════╡
-│ component ┆ foo        ┆ 101B ┆  100% ┆ foo       ┆ <root> │
-├╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┤
-│ module    ┆ unknown(0) ┆  39B ┆   39% ┆ -         ┆ foo    │
-╰───────────┴────────────┴──────┴───────┴───────────┴────────╯
+╭───────────┬────────────┬───────┬───────┬───────────┬────────╮
+│ KIND      ┆ NAME       ┆  SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT │
+╞═══════════╪════════════╪═══════╪═══════╪═══════════╪════════╡
+│ component ┆ foo        ┆ 101 B ┆  100% ┆ foo       ┆ <root> │
+├╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┤
+│ module    ┆ unknown(0) ┆  39 B ┆   39% ┆ -         ┆ foo    │
+╰───────────┴────────────┴───────┴───────┴───────────┴────────╯
 ╭──────────┬────────────────────╮
 │ KIND     ┆ VALUE              │
 ╞══════════╪════════════════════╡
--- a/tests/cli/metadata-component.wat.stdout
+++ b/tests/cli/metadata-component.wat.stdout
@@ -1,12 +1,12 @@
-╭───────────┬───────────────────┬──────┬───────┬───────────┬─────────╮
-│ KIND      ┆ NAME              ┆ SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT  │
-╞═══════════╪═══════════════════╪══════╪═══════╪═══════════╪═════════╡
-│ component ┆ my-name           ┆ 136B ┆  100% ┆ -         ┆ <root>  │
-├╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ module    ┆ submodule         ┆  27B ┆   20% ┆ -         ┆ my-name │
-├╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
-│ module    ┆ another submodule ┆  35B ┆   26% ┆ -         ┆ my-name │
-╰───────────┴───────────────────┴──────┴───────┴───────────┴─────────╯
+╭───────────┬───────────────────┬───────┬───────┬───────────┬─────────╮
+│ KIND      ┆ NAME              ┆  SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT  │
+╞═══════════╪═══════════════════╪═══════╪═══════╪═══════════╪═════════╡
+│ component ┆ my-name           ┆ 136 B ┆  100% ┆ -         ┆ <root>  │
+├╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
+│ module    ┆ submodule         ┆  27 B ┆   20% ┆ -         ┆ my-name │
+├╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┤
+│ module    ┆ another submodule ┆  35 B ┆   26% ┆ -         ┆ my-name │
+╰───────────┴───────────────────┴───────┴───────┴───────────┴─────────╯
 ╭───────┬────────────────────────────╮
 │ KIND  ┆ VALUE                      │
 ╞═══════╪════════════════════════════╡
--- a/tests/cli/metadata.wat.stdout
+++ b/tests/cli/metadata.wat.stdout
@@ -1,7 +1,7 @@
 ╭────────┬────────────┬──────┬───────┬───────────┬────────╮
 │ KIND   ┆ NAME       ┆ SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT │
 ╞════════╪════════════╪══════╪═══════╪═══════════╪════════╡
-│ module ┆ unknown(0) ┆   8B ┆  100% ┆ -         ┆ <root> │
+│ module ┆ unknown(0) ┆  8 B ┆  100% ┆ -         ┆ <root> │
 ╰────────┴────────────┴──────┴───────┴───────────┴────────╯
 ╭───────┬───────────╮
 │ KIND  ┆ VALUE     │
--- a/tests/cli/add-metadata-clear.wat.stdout
+++ b/tests/cli/add-metadata-clear.wat.stdout
@@ -1,8 +1,8 @@
-╭────────┬────────────┬──────┬───────┬───────────┬────────╮
-│ KIND   ┆ NAME       ┆ SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT │
-╞════════╪════════════╪══════╪═══════╪═══════════╪════════╡
-│ module ┆ unknown(0) ┆ 110B ┆  100% ┆ bar       ┆ <root> │
-╰────────┴────────────┴──────┴───────┴───────────┴────────╯
+╭────────┬────────────┬───────┬───────┬───────────┬────────╮
+│ KIND   ┆ NAME       ┆  SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT │
+╞════════╪════════════╪═══════╪═══════╪═══════════╪════════╡
+│ module ┆ unknown(0) ┆ 110 B ┆  100% ┆ bar       ┆ <root> │
+╰────────┴────────────┴───────┴───────┴───────────┴────────╯
 ╭──────────────┬────────────────────╮
 │ KIND         ┆ VALUE              │
 ╞══════════════╪════════════════════╡
--- a/tests/cli/add-metadata-merge-sections.wat.stdout
+++ b/tests/cli/add-metadata-merge-sections.wat.stdout
@@ -1,7 +1,7 @@
 ╭────────┬────────────┬──────┬───────┬───────────┬────────╮
 │ KIND   ┆ NAME       ┆ SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT │
 ╞════════╪════════════╪══════╪═══════╪═══════════╪════════╡
-│ module ┆ unknown(0) ┆  54B ┆  100% ┆ foo, bar  ┆ <root> │
+│ module ┆ unknown(0) ┆ 54 B ┆  100% ┆ foo, bar  ┆ <root> │
 ╰────────┴────────────┴──────┴───────┴───────────┴────────╯
 ╭──────────┬───────────╮
 │ KIND     ┆ VALUE     │
--- a/tests/cli/add-metadata-overwrite-name.wat.stdout
+++ b/tests/cli/add-metadata-overwrite-name.wat.stdout
@@ -1,7 +1,7 @@
 ╭────────┬──────┬──────┬───────┬───────────┬────────╮
 │ KIND   ┆ NAME ┆ SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT │
 ╞════════╪══════╪══════╪═══════╪═══════════╪════════╡
-│ module ┆ foo  ┆  21B ┆  100% ┆ -         ┆ <root> │
+│ module ┆ foo  ┆ 21 B ┆  100% ┆ -         ┆ <root> │
 ╰────────┴──────┴──────┴───────┴───────────┴────────╯
 ╭───────┬───────────╮
 │ KIND  ┆ VALUE     │
--- a/tests/cli/add-metadata.wat.stdout
+++ b/tests/cli/add-metadata.wat.stdout
@@ -1,7 +1,7 @@
 ╭────────┬──────┬──────┬───────┬───────────┬────────╮
 │ KIND   ┆ NAME ┆ SIZE ┆ SIZE% ┆ LANGUAGES ┆ PARENT │
 ╞════════╪══════╪══════╪═══════╪═══════════╪════════╡
-│ module ┆ foo  ┆  84B ┆  100% ┆ bar       ┆ <root> │
+│ module ┆ foo  ┆ 84 B ┆  100% ┆ bar       ┆ <root> │
 ╰────────┴──────┴──────┴───────┴───────────┴────────╯
 ╭──────────────┬────────────╮
 │ KIND         ┆ VALUE      │
